### PR TITLE
Delete multiple elements

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java [hunk 1]
```diff
@@ -204,6 +204,17 @@ public void deleteElement(UUID id, String userId) {
         }
     }
 
+    public void deleteElements(List<UUID> uuids, String userId) {
+        try {
+            uuids.forEach(id -> directoryService.deleteElement(id, userId));
+            directoryService.deleteDirectoryElements(uuids, userId);
+            // FIXME dirty fix to ignore errors and still delete the elements in the directory-server. To delete when handled properly.
```
Comment (Meklo) [inline src/main/java/org/gridsuite/explore/server/services/ExploreService.java:211]: What prevents us to handle this properly at this time ?

File: src/main/java/org/gridsuite/explore/server/ExploreController.java [hunk 2]
```diff
@@ -242,6 +242,16 @@ public ResponseEntity<Void> deleteElement(@PathVariable("elementUuid") UUID elem
         return ResponseEntity.ok().build();
     }
 
+    @DeleteMapping(value = "/explore/elements/{directoryUuid}/delete-stashed")
+    @Operation(summary = "Remove directories/elements")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "Filter creation request delegated to filter server")})
```
Comment (achour94) [inline src/main/java/org/gridsuite/explore/server/ExploreController.java:247]: That is not the right description

File: src/main/java/org/gridsuite/explore/server/ExploreController.java [hunk 3]
```diff
@@ -242,6 +242,16 @@ public ResponseEntity<Void> deleteElement(@PathVariable("elementUuid") UUID elem
...
+    public ResponseEntity<Void> deleteElements(@RequestParam("ids") List<UUID> elementsUuid,
+                                               @RequestHeader("userId") String userId,
+                                               @PathVariable String directoryUuid) {
```
Comment (achour94) [inline src/main/java/org/gridsuite/explore/server/ExploreController.java:250]: This variable is not used

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java [hunk 4]
```diff
@@ -204,6 +204,17 @@ public void deleteElement(UUID id, String userId) {
...
+        } catch (Exception e) {
+            LOGGER.error(e.toString(), e);
+            directoryService.deleteDirectoryElements(uuids, userId);
```
Comment (achour94) [inline src/main/java/org/gridsuite/explore/server/services/ExploreService.java:214]: Why this method is called in both try and catch !?

**Review by achour94 — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java (hunk 1 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:211]: I used the same logic of this PR: https://github.com/gridsuite/explore-server/pull/27. we ignore the exception and do the deletion anyway

File: src/main/java/org/gridsuite/explore/server/ExploreController.java (hunk 3 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/explore/server/ExploreController.java:250]: Yes, but we must have an uuid at this position or the gateway will reject our fetch with 403 status

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java (hunk 4 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:214]: I used the same logic of this PR: https://github.com/gridsuite/explore-server/pull/27. we ignore the exception and do the deletion anyway

File: src/main/java/org/gridsuite/explore/server/ExploreController.java (hunk 2 above)
Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/explore/server/ExploreController.java]: fixed it

File: src/main/java/org/gridsuite/explore/server/services/ExploreService.java (hunk 4 above)
Comment (achour94) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:214]: consider this suggestion 
public void deleteElements(List<UUID> uuids, String userId) {
        try {
            uuids.forEach(id -> directoryService.deleteElement(id, userId));
            // FIXME dirty fix to ignore errors and still delete the elements in the directory-server. To delete when handled properly.
        } catch (Exception e) {
            LOGGER.error(e.toString(), e);
        } finally {
            directoryService.deleteDirectoryElements(uuids, userId);
        }
    }

Comment (YenguiSeddik) [reply src/main/java/org/gridsuite/explore/server/services/ExploreService.java:214]: done

**Review by achour94 — APPROVED** [review]

**Review by Meklo — APPROVED** [review]

**Pull request merged**